presentation/src/main/java/com/anwera64/pagodividido/main/MainView.kt

```diff
@@ -0,0 +1,155 @@
+@Composable
+fun TripsContent(
+    trips: List<TripModel>,
+    createNewTripAction: () -> Unit,
+    onTripSelected: (uid: String, name: String) -> Unit
+) {
+    Scaffold(
+        topBar = {
+            TopAppBar(
+                title = {
+                    Text(text = stringResource(id = R.string.title_trips))
+                },
+            )
+        },
+        floatingActionButton = {
+            FloatingActionButton(onClick = createNewTripAction) {
+                Icon(
+                    imageVector = Icons.Default.Add,
+                    contentDescription = stringResource(id = R.string.add_new_trip)
+                )
+            }
+        },
+        floatingActionButtonPosition = FabPosition.End,
+        modifier = Modifier.fillMaxSize()
+    ) { contentPadding ->
+        TripList(trips, onTripSelected, Modifier.padding(contentPadding))
+    }
+}
+
+
+@Composable
+private fun TripList(
+    trips: List<TripModel>,
+    onTripSelected: (uid: String, name: String) -> Unit,
+    modifier: Modifier = Modifier
+) {
+    val scrollState = rememberLazyListState()
+    LazyColumn(modifier = modifier.fillMaxSize(), state = scrollState) {
+        items(trips) { tripModel ->
+            TripItem(
+                tripModel = tripModel,
+                modifier = Modifier.clickable {
+                    onTripSelected(tripModel.uid, tripModel.name)
+                }
+            )
+        }
+    }
+}
+
+@Composable
+private fun TripItem(tripModel: TripModel, modifier: Modifier) {
+    Row(
+        modifier = modifier
+            .fillMaxWidth()
+            .background(MaterialTheme.colorScheme.background)
+            .padding(horizontal = 24.dp, vertical = 8.dp),
+        horizontalArrangement = Arrangement.SpaceBetween,
+        verticalAlignment = Alignment.CenterVertically
+    ) {
+        Column() {
+            Text(
+                text = tripModel.name,
+                color = MaterialTheme.colorScheme.primary,
+                style = MaterialTheme.typography.titleMedium,
+                fontWeight = FontWeight.Bold
+            )
+            Text(
+                text = "${stringResource(id = R.string.amount_spent)} ${tripModel.totalSpent}",
+                color = MaterialTheme.colorScheme.secondary,
+                style = MaterialTheme.typography.bodySmall
+            )
+            Text(
+                text = buildCompanionString(tripModel),
+                color = MaterialTheme.colorScheme.tertiary,
+                style = MaterialTheme.typography.labelMedium
+            )
+        }
+        Image(
+            painter = painterResource(id = R.drawable.ic_chevron_right),
+            contentDescription = null,
+            colorFilter = ColorFilter.tint(MaterialTheme.colorScheme.primary)
+        )
+    }
+}
+
+private fun buildCompanionString(tripModel: TripModel): String {
+    var string = String()
+    tripModel.companions.forEachIndexed { i, companion ->
```

Comment:
we can do a joinToString function, in this case we're going to miss the "and" part, but is not relevant I think

Reply:
Didn't know about that. I'll take a look at it, thanks!